objectivec/ort_checkpoint_internal.h

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "ort_checkpoint.h"
+
+#import "cxx_api.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface ORTCheckpoint ()
+
+- (Ort::CheckpointState&)CXXAPIOrtCheckpoint;
+
+- (nullable instancetype)initWithCXXAPIOrtCheckPointFromPath:(NSString*)path
```

Comment:
I would just call it `initWithPath:error`. I wonder if we can make it public and replace `loadCheckpointFromPath:error`?

@baijumeswani is there a reason to prefer a static `LoadCheckpoint()` method over an ORTCheckpoint constructor that takes a path?